integ/workflow/wf_state_api_fail_and_proceed/routers.go

```diff
@@ -0,0 +1,54 @@
+package wf_state_api_fail_and_proceed
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/indeedeng/iwf/gen/iwfidl"
+	"github.com/indeedeng/iwf/integ/workflow/common"
+)
+
+const (
+	WorkflowType = "wf_state_api_fail_and_proceed"
+	State1       = "S1"
+)
+
+type handler struct {
+	invokeHistory map[string]int64
+	invokeData    map[string]interface{}
+}
+
+func NewHandler() common.WorkflowHandler {
+	return &handler{
+		invokeHistory: make(map[string]int64),
+		invokeData:    make(map[string]interface{}),
+	}
+}
+
+func (h *handler) ApiV1WorkflowStateStart(c *gin.Context) {
+	var req iwfidl.WorkflowStateStartRequest
+	if err := c.ShouldBindJSON(&req); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	log.Println("received state start request, ", req)
+
+	if req.GetWorkflowType() == WorkflowType {
+		h.invokeHistory[req.GetWorkflowStateId()+"_start"]++
+		if req.GetWorkflowStateId() == State1 {
+			c.JSON(http.StatusBadRequest, iwfidl.WorkflowStateStartResponse{})
+			return
+		}
+	}
+
+	panic("should not get here")
+}
+
+func (h *handler) ApiV1WorkflowStateDecide(c *gin.Context) {
```

Comment:
it's better to add assertion here that the context has the correct result about start api failure (otherwise just panic)